docs/HandlingErrors.md

```diff
@@ -0,0 +1,33 @@
+# Handling errors
+
+## Background
+
+Many different errors can happen at various different points. Having a system to effectively present errors and the details needed to know what to do, and when, is critically important
+
+## Error situations
+
+1. Something broke, and it might be transient, you can try that action again
+1. Some 3rd party dependency failed, and you’ll need to figure out what to do (e.g. GitHub is down, or Xcode couldn't be installed)
+1. You hit a fastlane.ci or fastlane bug
+
+Some errors are recoverable by just retrying (we do that automatically for things that make sense). Some are recoverable only when somebody logs into the box and fixes the problem. Some problems are not recoverable at all.
+
+## Presenting Errors
+
+Current thinking about errors, they:
+
+1. Show up and stay around until dismissed on a given page, e.g. fastlane broke on this build, but if you want, you can retry it, but you only want this to persist on the project build page
```

Comment:
Is this a notification for an error wherever you are in the UI? Or just how it's displayed in the build details?

Reply:
So the assumption is that if there is a CI error on one build it is a problem for all builds? If that's true this sounds good to me.

Reply:
Yup, while it's not guaranteed, it's most likely that a runner problem is a problem for all.

Reply:
Can we add this assumption to the doc?